geo/src/algorithm/simplify_vw.rs

```diff
@@ -110,17 +111,18 @@ where
             current: i + 1,
             left: i,
             right: i + 2,
-            intersector: (),
+            intersector: false,
         })
-        .collect::<BinaryHeap<VScore<T, ()>>>();
+        .collect::<BinaryHeap<VScore<T, bool>>>();
     // While there are still points for which the associated triangle
     // has an area below the epsilon
     while let Some(smallest) = pq.pop() {
-        // This triangle's area is above epsilon, so skip it
         if smallest.area > *epsilon {
-            continue;
+            // no need to keep trying: the min-heap ensures that we process triangles in order
+            // so if we see one that exceeds the tolerance we're done: everything else is too big
+            break;
```

Comment:
I'm not very familiar with this algorithm, but is it true that we can guarantee no smaller triangles will be added in subsequent calls to `recompute_triangles`?

Reply:
Right, `pq` is a min heap, so we will always get the smallest triangle *currently* in the min heap.

But then we add more triangles to the head on line 142.
`recompute_triangles(&smallest, orig, &mut pq, ll, left, right, rr, max, epsilon);`

Is it possible that by `continuing` instead of `breaking`, we'd add triangles on line 142 that would be worth considering, or is there some property of this algorithm that ensures subsequent calls to `recompute_triangles` will not add such triangles?

Reply:
> Is it possible that by continuing instead of breaking, we'd add triangles on line 142 that would be worth considering, or is there some property of this algorithm that ensures subsequent calls to recompute_triangles will not add such triangles?

I don't think so: once we hit the first triangle above epsilon we never even make it as far as line 125 again, let alone line 142, because `continue` restarts the loop with a new triangle, which is immediately discarded.

Reply:
Oh right, thanks for walking me through that.

Reply:
I keep going back to check my assumptions, but I'm pretty happy that this logic is correct, and also borne out elsewhere:
- [This is the original implementation, which uses `break`](https://github.com/huonw/isrustfastyet/blob/25e7a68ff26673a8556b170d3c9af52e1c818288/mem/line_simplify.rs#L54-L55)
- [A Go implementation](https://github.com/paulmach/orb/blob/master/simplify/visvalingam.go#L93)